File: bubble-sort/bsort.js
function bubbleSort(coll) {
    const sorted = Array.from(coll);
    const len = sorted.length;
    if (len < 2) {
        return sorted;
    }
    let swapped;
    do {
        swapped = false;
        for (let i = 0; i <= (len - 2); ++i) {
            if (sorted[i] > sorted[i + 1]) {
                [sorted[i], sorted[i + 1]] = [sorted[i + 1], sorted[i]];
                swapped = true;
            }
        }
    } while (swapped);
    return sorted;
}

const collection = [5, 1, 4, 7, 2, 12, 6, 8];
console.log(bubbleSort(collection));

File: bubble-sort/bsort.go
package main

import "fmt"

func bubbleSort(coll []int) []int {
    result := make([]int, len(coll))
    copy(result, coll)
    for {
        swapped := false
        for i := 0; i < (len(result) - 1); i++ {
            if result[i] > result[i + 1] {
                tmp := result[i]
                result[i] = result[i + 1]
                result[i + 1] = tmp
                swapped = true
            }
        }
        if !swapped {
            break
        }
    }
    return result
}

func main() {
    numbers := [...]int {3, 8, 1, 7, 9, 12, 6, 2}
    sorted := bubbleSort(numbers[:])
    fmt.Println(sorted)
}

